x-pack/plugins/cases/public/client/helpers/can_use_cases.ts

```diff
@@ -6,24 +6,64 @@
  */
 
 import type { ApplicationStart } from '@kbn/core/public';
-import { OBSERVABILITY_OWNER, SECURITY_SOLUTION_OWNER } from '../../../common/constants';
+import {
+  FEATURE_ID,
+  GENERAL_CASES_OWNER,
+  OBSERVABILITY_OWNER,
+  SECURITY_SOLUTION_OWNER,
+} from '../../../common/constants';
+import { getUICapabilities } from './capabilities';
+import { CasesPermissions } from '../../../common';
 
-export type CasesOwners = typeof SECURITY_SOLUTION_OWNER | typeof OBSERVABILITY_OWNER;
+export type CasesOwners =
+  | typeof SECURITY_SOLUTION_OWNER
+  | typeof OBSERVABILITY_OWNER
+  | typeof GENERAL_CASES_OWNER;
 
 /*
  * Returns an object denoting the current user's ability to read and crud cases.
  * If any owner(securitySolution, Observability) is found with crud or read capability respectively,
  * then crud or read is set to true.
  * Permissions for a specific owners can be found by passing an owner array
  */
-
 export const canUseCases =
   (capabilities: Partial<ApplicationStart['capabilities']>) =>
   (
-    owners: CasesOwners[] = [OBSERVABILITY_OWNER, SECURITY_SOLUTION_OWNER]
-  ): { crud: boolean; read: boolean } => ({
-    crud:
-      (capabilities && owners.some((owner) => capabilities[`${owner}Cases`]?.crud_cases)) ?? false,
-    read:
-      (capabilities && owners.some((owner) => capabilities[`${owner}Cases`]?.read_cases)) ?? false,
-  });
+    owners: CasesOwners[] = [OBSERVABILITY_OWNER, SECURITY_SOLUTION_OWNER, GENERAL_CASES_OWNER]
+  ): CasesPermissions => {
+    const aggregatedPermissions = owners.reduce<CasesPermissions>(
+      (acc, owner) => {
+        const userCapabilitiesForOwner = getUICapabilities(capabilities[getFeatureID(owner)]);
+
+        acc.create = acc.create || userCapabilitiesForOwner.create;
+        acc.read = acc.read || userCapabilitiesForOwner.read;
+        acc.update = acc.update || userCapabilitiesForOwner.update;
+        acc.delete = acc.delete || userCapabilitiesForOwner.delete;
+        acc.push = acc.push || userCapabilitiesForOwner.push;
+        const allFromAcc = acc.create && acc.read && acc.update && acc.delete && acc.push;
+        acc.all = acc.all || userCapabilitiesForOwner.all || allFromAcc;
+
+        return acc;
+      },
+      {
+        all: false,
+        create: false,
+        read: false,
+        update: false,
+        delete: false,
+        push: false,
+      }
+    );
+
+    return {
+      ...aggregatedPermissions,
+    };
+  };
+
+const getFeatureID = (owner: CasesOwners) => {
+  if (owner === GENERAL_CASES_OWNER) {
```

Comment:
The issue here is that we're assuming that the format will be `<feature>Cases`, but for cases in the stack that would result in `casesCases` instead of `generalCases` so we need to check for that specifically.